tsfc/fiatinterface.py

```diff
@@ -122,19 +123,30 @@ def _(element, vector_is_mixed):
         element = convert(element, vector_is_mixed)
         # Splat quadrilateral elements that are on TFEs back into
         # something with the correct entity dofs.
-        nodes = element.dual.nodes
-        ref_el = FiredrakeQuadrilateral()
-
-        entity_ids = element.dual.entity_ids
-        flat_entity_ids = {}
-        flat_entity_ids[0] = entity_ids[(0, 0)]
-        flat_entity_ids[1] = dict(enumerate(entity_ids[(0, 1)].values() +
-                                            entity_ids[(1, 0)].values()))
-        flat_entity_ids[2] = entity_ids[(1, 1)]
-
-        element.dual = DualSet(nodes, ref_el, flat_entity_ids)
-        element.ref_el = ref_el
-        return element
+
+        def cell_to_quad(element):
```

Comment:
I am also somewhat convinced that this is causing the problems I'm experiencing with the caching errors described in Firedrake issue #823. This was added to address some problems I experienced with `RTCF` elements on a `FiredrakeQuadrilateral`.

I managed to work out exactly which particular `pytest` parameters are having the caching issue: they all are tests related to `RTCF` elements. I think I need to discuss this in more detail with you to create a fix for this. Perhaps fixing this might solve my problem in Firedrake issue #823.